tests/repository/test_repository.py

```diff
@@ -566,6 +581,37 @@ def test_walk(repository, generate_directory):
     ]
 
 
+def test_copy_tree(repository, generate_directory, tmp_path_parametrized):
+    """Test the ``Repository.copy_tree`` method."""
+    directory = generate_directory({'file_a': None, 'relative': {'file_b': None, 'sub': {'file_c': None}}})
+    repository.put_object_from_tree(str(directory))
+
+    repository.copy_tree(tmp_path_parametrized)
```

Comment:
You are not testing when you provide a source path inside the node repo, no? Perhaps I would add a test for that. I wonder if it makes sense to have a cool parametrized fixture like you made for `Union[str, pathlib.Path]` but for `FilePath = Union[str, pathlib.PurePosixPath]` 🤔

Reply:
I will see how I can add the test.

Reply:
So you tagged me for re-review but I don't see this test added, did you decided not to do it? I mean, it is ok, I guess it is not super critical, I'm just checking it was intentional and not that you forgot or some commit didn't get pushed =).

Reply:
I have added the test though. It now parametrizes over `path` as follows:
```python
@pytest.mark.parametrize('path', ('.', 'relative'))
```
and `path` is passed both to `copy_tree` and then used in `walk` for the check. I didn't also parametrize each of these values over `str` and `pathlib.Path`.

Reply:
Totally right, my bad I missed that!